homeassistant/components/smartthings/__init__.py

```diff
@@ -3,415 +3,96 @@
+        ],
+        client=client,
+        scenes=scenes,
+    )
 
     await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
-    return True
-
 
-async def async_get_entry_scenes(entry: ConfigEntry, api):
-    """Get the scenes within an integration."""
-    try:
-        return await api.scenes(location_id=entry.data[CONF_LOCATION_ID])
-    except ClientResponseError as ex:
-        if ex.status == HTTPStatus.FORBIDDEN:
-            _LOGGER.exception(
-                (
-                    "Unable to load scenes for configuration entry '%s' because the"
-                    " access token does not have the required access"
-                ),
-                entry.title,
-            )
-        else:
-            raise
-    return []
+    return True
 
 
-async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+async def async_unload_entry(
+    hass: HomeAssistant, entry: SmartThingsConfigEntry
+) -> bool:
     """Unload a config entry."""
-    broker = hass.data[DOMAIN][DATA_BROKERS].pop(entry.entry_id, None)
-    if broker:
-        broker.disconnect()
-
     return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
-
-
-async def async_remove_entry(hass: HomeAssistant, entry: ConfigEntry) -> None:
-    """Perform clean-up when entry is being removed."""
-    api = SmartThings(async_get_clientsession(hass), entry.data[CONF_ACCESS_TOKEN])
-
-    # Remove the installed_app, which if already removed raises a HTTPStatus.FORBIDDEN error.
-    installed_app_id = entry.data[CONF_INSTALLED_APP_ID]
-    try:
-        await api.delete_installed_app(installed_app_id)
-    except ClientResponseError as ex:
-        if ex.status == HTTPStatus.FORBIDDEN:
-            _LOGGER.debug(
-                "Installed app %s has already been removed",
-                installed_app_id,
-                exc_info=True,
-            )
-        else:
-            raise
-    _LOGGER.debug("Removed installed app %s", installed_app_id)
-
-    # Remove the app if not referenced by other entries, which if already
-    # removed raises a HTTPStatus.FORBIDDEN error.
-    all_entries = hass.config_entries.async_entries(DOMAIN)
-    app_id = entry.data[CONF_APP_ID]
-    app_count = sum(1 for entry in all_entries if entry.data[CONF_APP_ID] == app_id)
-    if app_count > 1:
-        _LOGGER.debug(
-            (
-                "App %s was not removed because it is in use by other configuration"
-                " entries"
-            ),
-            app_id,
-        )
-        return
-    # Remove the app
-    try:
-        await api.delete_app(app_id)
-    except ClientResponseError as ex:
-        if ex.status == HTTPStatus.FORBIDDEN:
-            _LOGGER.debug("App %s has already been removed", app_id, exc_info=True)
-        else:
-            raise
-    _LOGGER.debug("Removed app %s", app_id)
-
-    if len(all_entries) == 1:
-        await unload_smartapp_endpoint(hass)
```

Comment:
I am wondering if we should remove the smartapp during the migration. Do we expect users to rollback?